accessor.go

```diff
@@ -54,11 +54,34 @@ func (sa *ShardAccessor) Shard() shard.Key {
 	return sa.shard.key
 }
 
+func (sa *ShardAccessor) Read(p []byte) (int, error) {
+	sa.lk.Lock()
+	defer sa.lk.Unlock()
+
+	if sa.mmapr == nil {
+		if f, ok := sa.data.(*os.File); ok {
+			if mmapr, err := mmap.Open(f.Name()); err != nil {
+				log.Warnf("failed to mmap reader of type %T: %s; using reader as-is", sa.data, err)
+			} else {
+				sa.mmapr = mmapr
+			}
+		}
+	}
```

Comment:
Let's extract this logic to a `tryMmap()` method (and document it requires the lock to be held) and refactor the `Blockstore()` method to use that?